### DISCUSSION TITLE
Questions related to qip

### DISCUSSION THREAD

**Discussion by purva-thakre** [qutip/qutip-qip#23] [Q&A]
@BoxiLi  What's a preferred way to import modules ?

There's no consistent style in all of the `qutip-qip` modules. In some cases, the absolute path has been specified whereas some use relative imports. Since I am making changes to how `Gate` class is imported, I can change imports for others as well. 

_BTW sorry for too many questions ! I though it would be better to start a discussion instead of creating issues for each question._

**Comment by BoxiLi** [comment]
> sorry for too many questions ! I though it would be better to start a discussion instead of creating issues for each question.

No worries, it is good to make thing clear in advance, to prevent unnecessary waste of time. Yeah for style discussion and code design, the discussion panel does seem to be a better place.

Actually, it is something open to discuss. I used to use absolute importation. But that caused some problem when moving from `qutip.qip`, to `qutip_qip` because I need to rename every imporation. So, my options are:
- For importation from `qutip`: we should only use the top namespace `qutip`, either `import qutip` or `from qutip import mesolve`. Things like `from qutip.qobj import Qobj` should be removed because `qutip_qip` now is a down stream package and in qutip v5 everything is moved around.
- For importation within `qutip_qip`: I don't really have a preference. For me, I'm using relative importation in all the newly written code, like `from .circuit import QubitCuit`. If you prefer others, it is also fine.
- For importation in `tests`: I think import from `import qutip_qip` or `from qutip_qip import ...` is better because we are testing an installed package.

Sorry that the style is a bit mixed. There is a mixture of 7-year-old code and 7-day-old code here :)

**Comment by rum1887** [comment]
hi, I get the following error when I try to build the doc!

<img width="755" alt="Screenshot 2023-05-28 at 7 14 30 PM" src="https://github.com/qutip/qutip-qip/assets/57267583/b0253644-22fe-4db7-bff1-c2937a8f5db7">

**Reply by hodgestar** [reply]
To build the docs you need to install the requirements listed in https://github.com/qutip/qutip-doc/blob/master/requirements.txt.

Probably https://qutip-qip.readthedocs.io/en/latest/contribution-docs.html#building-the-documentation should be updated to mention this.

Once you have the docs building, would you mind submitting a pull request to update the docs build documentation linked to above?

**Comment by hodgestar** [comment]
@rum1887 You posted this to a very old unrelated thread. I'm going to close this discussion so that others don't make the same mistake, but please feel free to open an new one or an issue if you need to.

**Reply by rum1887** [reply]
Sure!
